MekHQ/src/mekhq/campaign/autoResolve/helper/SetupTeams.java

```diff
@@ -0,0 +1,144 @@
+package mekhq.campaign.autoResolve.helper;
+
+import megamek.client.generator.RandomCallsignGenerator;
+import megamek.common.*;
+import megamek.common.alphaStrike.conversion.ASConverter;
+import megamek.common.annotations.Nullable;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.autoResolve.scenarioResolver.components.AutoResolveForce;
+import mekhq.campaign.mission.*;
+import mekhq.campaign.unit.Unit;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static mekhq.campaign.autoResolve.helper.SetupForces.getNewCrewRef;
+import static mekhq.campaign.force.StrategicFormation.getStandardForceSize;
+
+public class SetupTeams {
+
+    static public Map<Integer, List<AutoResolveForce>> setupTeams(Campaign campaign, List<Unit> units, AtBScenario scenario) {
+        Map<Integer, List<AutoResolveForce>> tmpTeams = new HashMap<>();
+        for (AutoResolveForce force : getAutoResolveForces(campaign, units, scenario)) {
+            tmpTeams.computeIfAbsent(force.team(), k -> new ArrayList<>()).add(force);
+        }
+        return tmpTeams;
+    }
+
+    static private List<AutoResolveForce> getAutoResolveForces(Campaign campaign, List<Unit> units, AtBScenario scenario) {
+        AtomicInteger i = new AtomicInteger(0);
+
+        List<AutoResolveForce> forces = new ArrayList<>();
+        forces.add(playerForce(i.incrementAndGet(), campaign.getPlayer().getName(), units));
+        forces.addAll(getBotForces(campaign, units, scenario, i));
+        return forces;
+    }
+
+    static private List<AutoResolveForce> getBotForces(Campaign campaign, List<Unit> units, AtBScenario scenario, AtomicInteger i) {
+        return new ArrayList<>(scenario.getBotForces().stream().map(botForce ->
+            new AutoResolveForce(botForce.getTeam(), botForce.getName(), i.get(),
+                setupBotEntities(campaign, units, botForce, scenario, i.getAndIncrement()))
+        ).toList());
+    }
+
+    static private AutoResolveForce playerForce(int playerID, String forceName, List<Unit> units) {
+        return new AutoResolveForce(1, forceName, playerID, units.stream().map(SetupTeams::breakEntityRef).filter(Objects::nonNull).toList());
+    }
+
+    @Nullable
+    public static Entity breakEntityRef(Unit unit) {
+        var entity = ASConverter.getUndamagedEntity(unit.getEntity());
```

Comment:
While it's not marked as such `getEntity()` can return `null`. Something to be aware of as I _believe_ if that occurs here it will result in an NPE.